psake.psm1

```diff
@@ -544,103 +544,105 @@ function CreateConfigurationForNewContext {
 }
 
 function ConfigureBuildEnvironment {
-    $framework = $psake.context.peek().config.framework
-    if ($framework -cmatch '^((?:\d+\.\d+)(?:\.\d+){0,1})(x86|x64){0,1}$') {
-        $versionPart = $matches[1]
-        $bitnessPart = $matches[2]
-    } else {
-        throw ($msgs.error_invalid_framework -f $framework)
-    }
-    $versions = $null
-    $buildToolsVersions = $null
-    switch ($versionPart) {
-        '1.0' {
-            $versions = @('v1.0.3705')
-        }
-        '1.1' {
-            $versions = @('v1.1.4322')
-        }
-        '2.0' {
-            $versions = @('v2.0.50727')
-        }
-        '3.0' {
-            $versions = @('v2.0.50727')
-        }
-        '3.5' {
-            $versions = @('v3.5', 'v2.0.50727')
-        }
-        '4.0' {
-            $versions = @('v4.0.30319')
-        }
-        {($_ -eq '4.5.1') -or ($_ -eq '4.5.2')} {
-            $versions = @('v4.0.30319')
-            $buildToolsVersions = @('14.0', '12.0')
-        }
-        {($_ -eq '4.6') -or ($_ -eq '4.6.1')} {
-            $versions = @('v4.0.30319')
-            $buildToolsVersions = @('14.0')
+    if (!(Get-Variable -Name IsWindows -ErrorAction SilentlyContinue) -or $IsWindows) {

```

Comment:
I wonder if it wouldn't be better to create some new functions: `ConfigureWindowsBuildEnvironment()`, `ConfigureLinuxBuildEnvironment()` and `ConfigureMacOSBuildEnvironment()`.  

Then have your `if` test call the Windows one, then add a test for $IsLinux and call the Linux one (ditto for $IsOSX).  For now, those functions could just write-warning that "build environment support for this platform has not been implemented yet.".